python/compiler/aiecc/main.py

```diff
@@ -685,32 +684,60 @@ async def process_host_cgen(self, aie_target, file_with_addresses):
                     cmd += [f"--gcc-toolchain={opts.sysroot}/usr"]
 
             install_path = aie.compiler.aiecc.configure.install_path()
+
+            # Setting everything up if linking against HSA
+            if opts.link_against_hsa:
+                cmd += ["-DHSA_RUNTIME"]
+                arch_name = opts.host_target.split("-")[0] + "-hsa"
+                hsa_path = os.path.join(aie.compiler.aiecc.configure.hsa_dir)
+                hsa_include_path = os.path.join(hsa_path, "..", "..", "..", "include")
+                hsa_lib_path = os.path.join(hsa_path, "..", "..")
+                hsa_so_path = os.path.join(hsa_lib_path, "libhsa-runtime64.so.1.9.0")
```

Comment:
does `-L<hsa_lib_path> -lhsa-runtime64` work here?

Reply:
For some reason when I do that I can compile the host code correctly but it complains it can't find the so. So this is why I add it to the rpath of the executable, but will change that to not specifically look for `libhsa-runtime64.so.1.9.0` and instead just look for `libhsa-runtime64.so`